router/src/components/form.rs

```diff
@@ -120,7 +120,10 @@ where
                                     Ok(url) => {
                                         request_animation_frame(move || {
                                             if let Err(e) = navigate(
-                                                &url.pathname,
+                                                &format!(
+                                                    "{}{}",
+                                                    url.pathname, url.search,
```

Comment:
If I'm reading this Stackoverflow Question right, I think we're supposed to append the hash if there was one on the original reference, and not append one if there's a hash in the URL.
https://stackoverflow.com/questions/2286402/url-fragment-and-302-redirects

Reply:
Agreed. Let's leave the hash out for now. The server request doesn't get the hash of course, so I'm trying to think... this would be "server function returns a redirect response with a hash on the URL" which I guess is possible but kind of odd. Can handle that in the future if we need to.

Reply:
So if I understood the quoted spec in the SO answer, there's a case where it would need to include a hash: when the original referring location includes a hash fragment and the new redirected location doesn't. In that case I think the redirected location is supposed to be appended with the same hash fragment.

> If the Location value provided in a 3xx (Redirection) response does not have a fragment component, a user agent MUST process the redirection as if the value inherits the fragment component of the URI reference used to generate the request target (i.e., the redirection inherits the original reference's fragment, if any). [1](https://www.rfc-editor.org/rfc/rfc7231#section-7.1.2)

Otherwise, the only case I can think of is as you stated: the server sends a new redirected location that includes a hash fragment.

Reply:
> So if I understood the quoted spec in the SO answer, there's a case where it would need to include a hash: when the original referring location includes a hash fragment and the new redirected location doesn't. In that case I think the redirected location is supposed to be appended with the same hash fragment.

This sounds correct to me, but (also having read it now) I think it applies to GET requests more than the POST requests we're handling here. So if I try to navigate to `/blog/posts/1#section-a` and it redirects me to `/blogs/post-1`, it should redirect to `/blog/posts-1#section-a`, because `#section-a` should refer to the same section of the page in either case. But when I'm doing something like POSTing a form, it doesn't necessarily make sense that if I post to `/create-todo#abc` (?) I should be redirected to `/todos/123#abc`.

So I'll merge this for now and when someone comes howling for blood about our failure to redirect  to the correct hash fragment they'll find this conversation and curse us all :-D